src/datumaro/cli/util/compare.py

```diff
@@ -83,22 +82,24 @@ def save(self, a: IDataset, b: IDataset):
         if len(a) != len(b):
             print("Datasets have different lengths: %s vs %s" % (len(a), len(b)))
 
-        a_classes = a.categories().get(AnnotationType.label, LabelCategories())
-        b_classes = b.categories().get(AnnotationType.label, LabelCategories())
-        class_mismatch = [
-            (idx, a_cls, b_cls)
-            for idx, (a_cls, b_cls) in enumerate(zip_longest(a_classes, b_classes))
-            if getattr(a_cls, "name", None) != getattr(b_cls, "name", None)
+        a_classes = set(
+            a_cls.name for a_cls in a.categories().get(AnnotationType.label, LabelCategories())
+        )
+        b_classes = set(
+            b_cls.name for b_cls in b.categories().get(AnnotationType.label, LabelCategories())
+        )
+
+        class_mismatch = [(diff, None) for diff in a_classes - b_classes] + [
```

Comment:
Is there any reason to keep `class_mismatch`? If not, we just do the following
```python
for idx, diff in enumerate(a_classes - b_classes):
     print(" #%s: > %s" % (idx, diff))
cnt = idx + 1
for idx, diff in enumerate(b_classes - a_classes):
     print(" #%s: < %s" % (idx + cnt, diff))
```

Reply:
Fix this like
```python
for idx, diff in enumerate(a_classes - b_classes):
    print(" #%s: > %s" % (idx, diff))

for idx, diff in enumerate(b_classes - a_classes, start=len(a_classes)):
    print(" #%s: < %s" % (idx, diff))
```

Reply:
Are they different, `len(a_classes - b_classes)` and `len(a_classes)`?

Reply:
Fix this in [b9538b0](https://github.com/openvinotoolkit/datumaro/pull/1205/commits/b9538b0125f1634c9f62c33b7e2b3a03fcac2f4b).